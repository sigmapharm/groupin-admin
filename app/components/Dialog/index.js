import React from 'react';
import Dialog from '@material-ui/core/Dialog/Dialog';
import DialogTitle from '@material-ui/core/DialogTitle/DialogTitle';
import DialogContent from '@material-ui/core/DialogContent/DialogContent';
import DialogActions from '@material-ui/core/DialogActions/DialogActions';
import Button from '@material-ui/core/Button/Button';
import Typography from '@material-ui/core/Typography/Typography';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';

export default ({
  title,
  open,
  onClose,
  onSubmit,
  children,
  showBtns = true,
  submitTitle,
  cancelTitle,
}) => (open ?  (
    <Dialog
      open={open}
      maxWidth="lg"
      onClose={onClose}
      scroll="paper"
      aria-labelledby="scroll-dialog-title"
    >
      <DialogTitle disableTypography>
        <Typography
          style={{
            display: 'flex',
            justifyContent: 'space-between',
            alignItems: 'center',
          }}
          variant="h5"
          color="primary"
        >
          <span>{title}</span>
          <IconButton color="primary" aria-label="Close" onClick={onClose}>
            <CloseIcon />
          </IconButton>
        </Typography>
      </DialogTitle>
      <DialogContent>{children}</DialogContent>
      {showBtns && (
        <DialogActions>
          <Button
            type="submit"
            variant="contained"
            color="primary"
            onClick={onClose}
          >
            {cancelTitle}
          </Button>
          <Button
            type="submit"
            variant="contained"
            color="primary"
            onClick={onSubmit}
          >
            {submitTitle}
          </Button>
        </DialogActions>
      )}
    </Dialog>
  ) : (
    <></>
  ))
