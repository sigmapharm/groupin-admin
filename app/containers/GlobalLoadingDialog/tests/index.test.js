import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CircularProgress from '@material-ui/core/CircularProgress';
import Dialog from '@material-ui/core/Dialog';
import { GlobalLoadingDialog } from '../index';

describe('<GlobalLoadingDialog />', () => {
  it('should match snapshot', () => {
    // given
    const props = {
      open: true,
      classes: {},
    };
    const shallowed = shallow(<GlobalLoadingDialog {...props} />);
    // then
    expect(shallowed.find(Dialog).length).toEqual(1);
    expect(shallowed.find(CircularProgress).length).toEqual(1);
    expect(toJson(shallowed)).toMatchSnapshot();
  });
});
