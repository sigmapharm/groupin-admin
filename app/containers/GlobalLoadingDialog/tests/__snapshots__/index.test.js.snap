// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GlobalLoadingDialog /> should match snapshot 1`] = `
<WithStyles(Dialog)
  PaperProps={
    Object {
      "classes": Object {
        "root": undefined,
      },
    }
  }
  fullWidth={true}
  open={true}
>
  <WithStyles(CircularProgress)
    color="primary"
    size={100}
  />
</WithStyles(Dialog)>
`;
